org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ResourceUtils.java

```diff
@@ -221,4 +221,17 @@ public static boolean isContainedIn(IPath location, Collection<IPath> paths) {
 		return false;
 	}
 
+	/**
+	 * Fix a path starting with ~/
```

Comment:
Expand paths starting with ~/ if necessary, or return the original path.

Reply:
I have updated the PR.